src/Factory/ShopBillingDataFactory.php

```diff
@@ -0,0 +1,37 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\RefundPlugin\Factory;
+
+use Sylius\Component\Resource\Exception\UnsupportedMethodException;
+use Sylius\RefundPlugin\Entity\ShopBillingData;
+use Sylius\RefundPlugin\Entity\ShopBillingDataInterface;
+
+final class ShopBillingDataFactory implements ShopBillingDataFactoryInterface
+{
+    public function createNew(): ShopBillingDataInterface
+    {
+        throw new UnsupportedMethodException('createNew');
```

Comment:
Exception message should be a little bit more explainable, something like: `To create ShopBillingData, you need to provide some arguments. Use ShopBillingDataFactoryInterface::createWithData instead`